map.go

```diff
@@ -235,7 +235,29 @@ func (m *Map) Lookup(key, valueOut interface{}) error {
 	if valueBytes == nil {
 		return nil
 	}
+	return m.valueProcessing(valueOut, valueBytes)
+}
+
+// LookupAndDelete retrieves and deletes a value from a Map.
+//
+// Calls Close() on valueOut if it is of type **Map or **Program,
+// and *valueOut is not nil.
+//
+// Returns an error if the key doesn't exist, see IsNotExist.
+func (m *Map) LookupAndDelete(key, valueOut interface{}) error {
+	valuePtr, valueBytes := makeBuffer(valueOut, m.fullValueSize)
+
+	if err := m.lookup(key, valuePtr); err != nil {
+		return err
+	}
 
+	if valueBytes == nil {
+		return nil
+	}
+	return m.valueProcessing(valueOut, valueBytes)
+}
+
+func (m *Map) valueProcessing(valueOut interface{}, valueBytes []byte) error {
 	if m.abi.Type.hasPerCPUValue() {
```

Comment:
Is it valid to call LookupAndDelete on a per-CPU map?

Reply:
As far as I can tell, it's valid. But maybe there is something, I don't know.